packages/plugin-paragraph/Paragraph.ts

```diff
@@ -17,4 +30,29 @@ export class Paragraph extends JWPlugin {
             return [context, parsingMap];
         }
     }
+    /**
+     * Change the formatting of the nodes in given range to Paragraph.
+     *
+     * @param params
+     */
+    applyParagraph(params: ParagraphParams): void {
+        let paragraph = params.nodeCreator(); // todo: make sure to keep attributes
+        if (paragraph.is(ParagraphNode)) {
+            const range = params.range || this.editor.vDocument.selection.range;
+            let nodesToConvert: VNode[];
+            if (range.isCollapsed()) {
+                nodesToConvert = [range.start.parent];
+            } else {
+                const selectedLeaves = range.selectedNodes(node => !node.children.length);
+                nodesToConvert = distinct(
+                    selectedLeaves.map(leaf => (leaf.atomic ? leaf.parent : leaf)),
+                );
+            }
+            for (const node of nodesToConvert) {
+                node.before(paragraph);
+                node.mergeWith(paragraph);
+                paragraph = params.nodeCreator();
+            }
+        }
+    }
```

Comment:
If we consider that `applyHeadingStyle` cannot exist without `Heading` installed, then this function can be removed. I mean, what would the toolbar look like if you only have Paragraph in the select and not Heading ? ^^'